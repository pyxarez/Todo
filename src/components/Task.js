import React from 'react';

import Button from './Button';
import './Task.res/style.css';

Task.propTypes = {
  isDone: React.PropTypes.bool.isRequired,
  title: React.PropTypes.string,
  handleDoneTaskClick: React.PropTypes.func.isRequired,
  handleEditTaskClick: React.PropTypes.func.isRequired
}

export default function Task({ isDone, title, handleDoneTaskClick, handleEditTaskClick }) {
  return (
    <div className="my-task-component">
      <input onChange={handleDoneTaskClick} checked={isDone} type="checkbox"/>
      <h2>{title}</h2>
      <Button onClick={handleEditTaskClick} type="edit" />
    </div>
  );
}