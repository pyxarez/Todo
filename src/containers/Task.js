import React, { Component, PropTypes } from 'react';
import { browserHistory } from 'react-router';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { checkDone } from '../store/actions/TaskListActions';
import { getProgress } from '../store/actions/ProgressActions';

import Task from '../components/Task';

export class TaskContainer extends Component {
  static propTypes = {
    categoryId: PropTypes.number.isRequired,
    task: PropTypes.shape({
      id: PropTypes.node.isRequired,
      title: PropTypes.string.isRequired,
      isDone: PropTypes.bool.isRequired,
      description: React.PropTypes.string.isRequired
    }),
    checkDone: PropTypes.func.isRequired,
    getProgress: React.PropTypes.func.isRequired,
  }

  handleEditTaskClick = () => {
    const categoryId = this.props.categoryId;
    const {
      id: taskId,
      title
    } = this.props.task;

    const path = `/edit/${categoryId}/${taskId}/${title}`;
    browserHistory.push(path);
  }

  handleDoneTaskClick = () => {
    const {
      checkDone,
      getProgress,
      categoryId
    } = this.props;
    const { id } = this.props.task;

    return checkDone(categoryId, id)
      .then(() => { getProgress(); });
  }

  render() {
    const {
      isDone,
      title
    } = this.props.task;

    return (
      <Task
        title={title}
        isDone={isDone}
        handleDoneTaskClick={this.handleDoneTaskClick}
        handleEditTaskClick={this.handleEditTaskClick}/>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    checkDone: bindActionCreators(checkDone, dispatch),
    getProgress: bindActionCreators(getProgress, dispatch)
  }
}

export default connect(null, mapDispatchToProps)(TaskContainer);