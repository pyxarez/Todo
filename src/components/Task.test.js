import React from 'react';
import { shallow } from 'enzyme';

import Task from './Task';

const setup = () => {
  const props = {
    isDone: true,
    title: 'Vassa',
    handleDoneTaskClick: () => {},
    handleEditTaskClick: () => {}
  }
  const wrapper = shallow(<Task {...props}/>)

  return {
    props,
    wrapper
  }
}

describe('Components :: Task', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});