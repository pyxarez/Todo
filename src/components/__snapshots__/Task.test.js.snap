exports[`Components :: Task should render self 1`] = `
<div
  className="my-task-component">
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox" />
  <h2>
    Vassa
  </h2>
  <Button
    onClick={[Function]}
    type="edit" />
</div>
`;
